### PR TITLE
Ft #163714458 user can log in

### PR DESCRIPTION
**Pull request by olamilekan000**
#### What does this PR do?
It creates a login protected endpoint.

#### Description of Task to be completed?
- setup jwt
- setup jwt strategy
- test

#### How should this be manually tested?
- Clone this branch into your machine using `git clone ft-user-can-log-in-163714458`
 - `cd` into the directory
- `npm i` to install all dependencies
- open postman
- send a POST request with the following data to the route below
POST   `localhost:8089/api/v1/auth/login`{
    "email": "chiz98@gmail.com",
    "password": "yourverysecretpassword"
  }

#### What are the relevant pivotal tracker stories?
- A user can log in
- https://www.pivotaltracker.com/story/show/163714458

### CODE REVIEW

File: Passport.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+passport.use(new JWTStrategy({
+	jwtFromRequest: ExtractJWT.fromHeader('authorization'),
+	secretOrKey: 'secret'	
+}, async (payload, done) => {
+	try{
+		const pool = new Pool();
+		const user = await DbQuery.getUserWithEmail(payload.email)
+		let dbRes = await pool.query(user);
+		await pool.end();
+		console.log(dbRes.rows[0])
```
Comment (veeqtor) [inline Passport.js:21]: @olamilekan000 Nice work, however, try not to push `console.log` to Github.

File: Passport.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import DbQuery from './server/model/';
+import { comparePwd } from './server/helpers/pwdHash';
+
+const JWTStrategy = passportJWT.Strategy;
+const ExtractJWT = passportJWT.ExtractJwt;
+const LocalStrategy = require('passport-local').Strategy;
+
+passport.use(new JWTStrategy({
+	jwtFromRequest: ExtractJWT.fromHeader('authorization'),
+	secretOrKey: 'secret'
```
Comment (veeqtor) [inline Passport.js:14]: You might want to take this out from your code and place it in a `.env` file

File: package.json [hunk 3]
```diff
@@ -23,11 +23,18 @@
   },
   "homepage": "https://github.com/olamilekan000/Politico#readme",
   "dependencies": {
```
Comment (veeqtor) [inline package.json:25]: @olamilekan000 , How you intend to start your app when you don't have nodemon installed?

File: server/controllers/signin.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { Pool } from 'pg';
+import token from '../helpers/jwt';
+import DbQuery from '../model';
+
+import { hashPwd } from '../helpers/pwdHash';
+
+export default class signInUser {
+  static async signin(req, res) {
```
Comment (veeqtor) [inline server/controllers/signin.js:8]: @olamilekan000 Can you explain to me what is going on here?

File: package.json (hunk 3 above)
Comment (olamilekan000) [reply package.json:25]: I have nodemon installed globally on my system

File: server/controllers/signin.js (hunk 4 above)
Comment (olamilekan000) [reply server/controllers/signin.js:8]: The hasPwd that was imported should be there and also I am exporting the signInUser class

File: Passport.js (hunk 1 above)
Comment (olamilekan000) [reply Passport.js:21]: Noted

File: Passport.js (hunk 2 above)
Comment (olamilekan000) [reply Passport.js:14]: Noted

**Pull request merged**